database/sql/src/main/scala/cromwell/database/slick/tables/MetadataEntryComponent.scala

```diff
@@ -88,26 +88,24 @@ trait MetadataEntryComponent {
 
   val metadataEntriesForJobKey = Compiled(
     (workflowExecutionUuid: Rep[String], callFullyQualifiedName: Rep[String], jobIndex: Rep[Option[Int]],
-     jobAttempt: Rep[Int]) => (for {
+     jobAttempt: Rep[Option[Int]]) => (for {
       metadataEntry <- metadataEntries
       if metadataEntry.workflowExecutionUuid === workflowExecutionUuid
       if metadataEntry.callFullyQualifiedName === callFullyQualifiedName
-      if (metadataEntry.jobIndex === jobIndex) ||
-        (metadataEntry.jobIndex.isEmpty && jobIndex.isEmpty)
-      if metadataEntry.jobAttempt === jobAttempt
+      if hasSameIndex(metadataEntry, jobIndex)
```

Comment:
I tried but `MetadataEntries` being in a trait I can't declare an implicit class in it, and I can't seem to be able to pull it out in an object either because it only exists inside the trait. I tried to add the methods directly in `MetadataEntries` itself but for some reason it breaks the tests so I'm going to leave it like this for this time I think :p